test/controllers/users_controller_test.rb

```diff
@@ -69,26 +69,63 @@ class UsersControllerTest < ActionController::TestCase
     assert_response(:not_found)
   end
 
+  test 'should soft-delete user profile' do
+    sign_in users(:global_admin)
+
+    try_soft_delete_user('profile', users(:standard_user))
+    @user = assigns(:user)
+
+    assert_response(:success)
+    assert_not_nil @user
+    assert @user.community_user.deleted
+  end
+
   test 'should soft-delete user' do
     sign_in users(:global_admin)
-    delete :soft_delete, params: { id: users(:standard_user).id, type: 'user' }
+
+    try_soft_delete_user('user', users(:standard_user))
+
     assert_response(:success)
     assert_not_nil assigns(:user)
-    assert_equal true, assigns(:user).deleted
+    assert assigns(:user).deleted
+  end
+
+  test 'only global moderators or admins should be able to soft-delete users' do
```

Comment:
Admins, global mods, and the user can soft-delete across the network (i.e. the whole account).  Local mods can soft-delete the community user only, not affecting the rest of the network.  Do/should we have tests that cover both of these variants?  (This would require that the test environment have more than one community.)